### ISSUE TITLE
Unable to fetch Job Object in Perform function

### ISSUE THREAD

**Issue by SteveBasik** [contribsys/faktory_worker_go#80]
I recently integrated Faktory into my GO application. I initialized a worker manager with the job type and the Perform function as mentioned below 

`mgr.Register("TicketGeneration", ReceiveJobs)`

In the **"ReceiveJobs"** function I only get access to the **"args"**. However, I am not able to fetch the **"Job"**. From the context I'm only able to fetch the **"Helper"** object.

```
func ReceiveJobs(ctx context.Context, args ...interface{}) {
      help := worker.HelperFor(ctx)
}
```

```
 Jid() string
 JobType() string
 Bid() string
 Batch(func(*client.Batch) error) error
 With(func(*client.Client) error) error
 TrackProgress(percent int, desc string, reserveUntil *time.Time) error
```
Is there a way that I can use the **"Jid"** from the helper object to fetch the job object ?

**Comment by mperham** [comment]
Nope. The job is an internal structure and not available to the worker. What are you trying to do?

**Comment by joe-wachirawit** [comment]
@mperham any update? I just want to do the same
cc. @SteveBasik

**Comment by mperham** [comment]
faktory_worker_go allows for middleware which run around the job execution and exposes the *Job object with retry count.

https://github.com/contribsys/faktory_worker_go/blob/main/middleware.go

I looked and it seems to be entirely undocumented so apologies if you didn't know about this API.
